.azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml

```diff
@@ -69,303 +75,302 @@ parameters:
-                : - checkout: ${{ checkoutRepository }}
-                    fetchDepth: 1 # the depth of commits to ask Git to fetch; if not set defaults to no limit
-                    path: 's/${{ checkoutRepository }}'
-
-          # [Agent] Prepare environment
-          #----------------------------
-          - task: PowerShell@2
-            displayName: 'Setup agent'
-            inputs:
-              targetType: inline
-              pwsh: true
-              script: |
-                # Load used functions
-                . (Join-Path '$(System.DefaultWorkingDirectory)' 'utilities' 'pipelines' 'sharedScripts' 'Set-EnvironmentOnAgent.ps1')
-
-                # Define PS modules to install on the runner
-                $Modules = @(
-                    @{ Name = 'Az.Accounts' },
-                    @{ Name = 'Az.Resources' }
-                )
-
-                # Additional PS modules need to be installed for the removal step in case it is enabled
-                if ('${{ parameters.removeDeployment}}' -eq 'true') {
-                  $Modules += @(
-                    @{ Name = 'Az.CognitiveServices' },
-                    @{ Name = 'Az.Compute' },
-                    @{ Name = 'Az.KeyVault' },
-                    @{ Name = 'Az.Monitor' },
-                    @{ Name = 'Az.OperationalInsights' },
-                    @{ Name = 'Az.RecoveryServices' }
-                  )
-                }
-
-                # Set agent up
-                Set-EnvironmentOnAgent -PSModules $Modules
-
-          # [Multi Repo] Support task
-          #--------------------------
-          - task: PowerShell@2
-            displayName: Handle Multi-Repo Invocation
-            inputs:
-              targetType: inline
-              pwsh: true
-              script: |
-                # Handle multiple-repositories
-                if( "${{ join(';',parameters.checkoutRepositories) }}".length -gt 0) {
-                  Write-Verbose "Multi-Repo Checkout" -Verbose
-                  $moduleRepoRoot = Join-Path '$(System.DefaultWorkingDirectory)' '$(modulesRepository)'
-                  $parametersRepoRoot = Join-Path '$(System.DefaultWorkingDirectory)' '${{ parameters.parametersRepository }}'
-                } else {
-                  Write-Verbose "No Multi-Repo Checkout" -Verbose
-                  $moduleRepoRoot = '$(System.DefaultWorkingDirectory)'
-                  $parametersRepoRoot = '$(System.DefaultWorkingDirectory)'
+  - job: deploy
+    displayName: '${{ parameters.jobDisplayName }}'
+    timeoutInMinutes: ${{ parameters.defaultJobTimeoutInMinutes }}
+    pool:
+      ${{ if ne(parameters.vmImage, '') }}:
+        vmImage: ${{ parameters.vmImage }}
+      ${{ if ne(parameters.poolName, '') }}:
+        name: ${{ parameters.poolName }}
+    ${{ if not( contains( parameters.templateFilePath, '.dependencies')) }}:
+      dependsOn:
+        - getParameterFilePaths
```

Comment:
Suggestion: Reference `jobs.getParameterFiles.yml` in this template instead of the stages template

Reply:
done